### PR TITLE
Test suite clean-up: Tests in `core` and `data`

### LINKED ISSUE #509: Test suite cleanup
When we migrated from `unittest` to `pytest`, our understanding was that we had to use only test functions rather than test cases with test methods. This lead to structures like this:

https://github.com/pystiche/pystiche/blob/65f4d787e44b1ffbf7e5b6e48298ed8c7460e5a9/tests/integration/enc/test_multi_layer_encoder.py#L50-L68

In general, there are quite a few places were the test suite looks like

```python
def test_Foo_bar():
    ...

def test_Foo_baz():
    ...
```

instead of 

```python
class TestFoo:
    def test_bar(self):
        ...

    def test_baz(self):
        ...
```

Cleaning this up is a little bit of busy work, but can be a good entry point to the development of `pystiche`.

### PR DESCRIPTION
**Pull request by DevPranjal**
Partially fixes #509 
Note: The diffs automatically detected by git seem to be very messed up.

### CODE REVIEW

**Comment by DevPranjal** [comment]
Hi @pmeier, this is a draft PR for a few tests. Could you please verify if I'm making the changes in the right direction?

File: tests/integration/core/test_utils.py [hunk 1]
```diff
@@ -8,61 +8,61 @@
...
-    batch_size = 2
-    num_channels = 1
-    size = 64
-    patch_size = 3
-    stride = 2
-
-    num_patches_per_dim = int((size - patch_size) // stride + 1)
-    torch.manual_seed(0)
-    x = torch.rand(batch_size, num_channels, *[size] * dims)
+class TestUtils:
```
Comment (pmeier) [inline tests/integration/core/test_utils.py:11]: Same as above. The module already namespaces this into `test_utils.py`.

File: tests/integration/data/test_collections.py [hunk 2]
```diff
@@ -13,246 +13,234 @@
...
-def test_DownloadableImage_download_exist(
-    tmpdir, test_image_url, test_image_file, test_image
-):
-    image = data.DownloadableImage(test_image_url)
-    file = path.join(tmpdir, image.file)
-
-    shutil.copyfile(test_image_file, file)
-
-    with pytest.raises(FileExistsError):
+class DownloadableImage:
```
Comment (pmeier) [inline tests/integration/data/test_collections.py:16]:
```suggestion
class TestDownloadableImage:
```

File: tests/integration/data/test_datasets.py [hunk 3]
```diff
@@ -47,48 +47,47 @@ def test_walkupto_depth(image_folder):
...
-
-    dataset = data.ImageFolderDataset(
-        image_folder, transform=transform, importer=importer
-    )
-
-    actual = dataset[0]
-    desired = path.join(image_folder, "transformed_imported_image0.jpg")
-    assert actual == desired
+class TestImageFolderDataset:
+    def test_image_folder_dataset(self, image_folder):
```
Comment (pmeier) [inline tests/integration/data/test_datasets.py:51]: We need a better naming scheme for this case to no repeat information. Maybe `test_core`?

**Review by pmeier — CHANGES_REQUESTED** [review]
Thanks for the PR @DevPranjal! Yes, in general you are moving in the right direction. I have some comments inline, but they should be fairly easy to resolve. 

The PR is already quite big and hard to review. So please address my comments and send a another PR for other files as a follow-up.

**Comment by pmeier** [comment]
Test failure is real:

```
=================================== FAILURES ===================================
______ TestDownloadableImageCollection.test_downloadable_image_collection ______
tests/integration/data/test_collections.py:116: in test_downloadable_image_collection
    collection.download(root=tmpdir)
pystiche/data/collections/download.py:200: in download
    image.download(root=root, overwrite=overwrite)
pystiche/data/collections/download.py:119: in download
    file = path.join(root, file)
/opt/hostedtoolcache/Python/3.6.14/x64/lib/python3.6/posixpath.py:80: in join
    a = os.fspath(a)
E   TypeError: expected str, bytes or os.PathLike object, not TestDownloadableImageCollection
```

Plus, the linting workflow also fails. Please run `tox -f lint` before you commit or install the `pre-commit` hooks so this will be done automatically.

File: tests/integration/data/test_datasets.py (hunk 3 above)
Comment (DevPranjal) [reply tests/integration/data/test_datasets.py:51]: There are quite a few places where this situation occurs, shall I change all these occurrences to `test_core` or maybe `test_main`.

File: tests/integration/data/test_licenses.py [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
-        data.NoLicense(),
-        data.PublicDomainLicense(),
-        data.ExpiredCopyrightLicense(1970),
-        data.PixabayLicense(),
-        data.CreativeCommonsLicense(("by", "sa"), "3.0"),
-    )
-    for license in licenses:
-        with subtests.test(license=license):
-            assert isinstance(repr(license), str)
+class TestLicenses:
```
[the remark this answers is not in the document]
Comment (DevPranjal) [reply tests/integration/data/test_licenses.py:4]: Couldn't think of a class name here. Any suggestions?

File: tests/integration/data/test_datasets.py (hunk 3 above)
Comment (pmeier) [reply tests/integration/data/test_datasets.py]: I like `test_main` better. And yes, please apply this everywhere.

File: tests/integration/data/test_licenses.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (pmeier) [reply tests/integration/data/test_licenses.py:4]: No need for a test class here. Just revert the changes in this file, and we are good to go.

**Review by pmeier — APPROVED** [review]
> Note: The diffs automatically detected by git seem to be very messed up.

Yeah, it is quite hard to review this. Given the fact that the coverage stays equal, I thin everything should be good to merge. Thanks a lot @DevPranjal!

**Pull request merged**